src/wallet.ts

```diff
@@ -1,230 +1,281 @@
 import axios from 'axios';
-import { ECPair, networks, payments, Psbt, confidential } from 'liquidjs-lib';
-//Libs
-import { coinselect } from './utils';
-//Types
-import { ECPairInterface } from 'liquidjs-lib/types/ecpair';
-import { Network } from 'liquidjs-lib/types/networks';
-
-export interface WatchOnlyWalletInterface {
-  address: string;
-  script: string;
+import {
+  Network,
+  networks,
+  payments,
+  address,
+  Psbt,
+  confidential,
+  Transaction,
+} from 'liquidjs-lib';
+import { fromAssetHash, toAssetHash } from './utils';
+import { AddressInterface } from './identity';
+
+/**
+ * Wallet abstraction.
+ */
+export interface WalletInterface {
   network: Network;
+  addresses: AddressInterface[];
+  blindingByScript: Record<string, Buffer>;
+  createTx(): string;
   updateTx(
-    psbtBase64: string,
-    inputs: Array<any>,
+    psetBase64: string,
+    unspents: Array<UtxoInterface>,
     inputAmount: number,
     outputAmount: number,
     inputAsset: string,
-    outputAsset: string
-  ): string;
+    outputAsset: string,
+    outputAddress: AddressInterface,
+    changeAddress: AddressInterface
+  ): any;
 }
 
-export class WatchOnlyWallet implements WatchOnlyWalletInterface {
+/**
+ * Implementation of Wallet Interface.
+ * @member network type of network (regtest...)
+ * @member addresses list of AddressInterface.
+ * @member blindingByScript a map scriptPubKey --> blindingPrivateKey.
+ * @method createTx init empty PSET.
+ * @method updateTx update a PSET with outputs and inputs (for Swap tx).
+ */
+export class Wallet implements WalletInterface {
   network: Network;
-  address: string;
-  script: string;
-  constructor({ address, network }: { address: string; network: Network }) {
-    const payment = payments.p2wpkh({ address, network });
+  addresses: AddressInterface[] = [];
```

Comment:
the member `addresses` is not used: only for creating blindingKeyByScript in the constructor.

Reply:
Yes it is externally, in marketOrderRequest for example to retrieve the addresses to fetch utxos from